services/google-sheets.ts

```diff
@@ -0,0 +1,231 @@
+import { GoogleSpreadsheet } from "google-spreadsheet";
+import { JWT } from "google-auth-library";
+import type {
+  GoogleSheetsConfig,
+  PitchSubmissionData,
+  RequestAssistanceData,
+  GoogleSheetsResponse,
+  GoogleSheetsValidationResponse,
+} from "../types/google-sheets";
+
+class GoogleSheetsService {
+  private readonly config: GoogleSheetsConfig;
+  private doc: GoogleSpreadsheet | null = null;
+
+  constructor() {
+    this.config = {
+      serviceAccountEmail: process.env.GOOGLE_SERVICE_ACCOUNT_EMAIL ?? "",
+      privateKey: process.env.GOOGLE_PRIVATE_KEY ?? "",
+      sheetId: process.env.GOOGLE_SHEETS_SHEET_ID ?? "",
+      ownerEmail: process.env.GOOGLE_SHEETS_OWNER_EMAIL ?? "",
+    };
+
+    if (
+      !this.config.serviceAccountEmail ||
+      !this.config.privateKey ||
+      !this.config.sheetId
+    ) {
+      throw new Error(
+        "Google Sheets configuration is incomplete. Please check environment variables."
+      );
+    }
+  }
+
+  private async initializeDoc(): Promise<GoogleSpreadsheet> {
+    if (this.doc) {
+      return this.doc;
+    }
+
+    try {
+      // Initialize JWT service account authentication
+      const serviceAccountAuth = new JWT({
+        email: this.config.serviceAccountEmail,
+        key: this.config.privateKey,
+        scopes: ["https://www.googleapis.com/auth/spreadsheets"],
+      });
+
+      // Initialize the Google Sheets document with service account authentication
+      this.doc = new GoogleSpreadsheet(this.config.sheetId, serviceAccountAuth);
+
+      // Load document info
+      await this.doc.loadInfo();
+
+      return this.doc;
+    } catch (error) {
+      console.error("Failed to initialize Google Sheets document:", error);
+      throw new Error(
+        "Failed to connect to Google Sheets. Please check your configuration."
+      );
+    }
+  }
+
+  private async getOrCreateSheet(sheetName: string): Promise<any> {
+    const doc = await this.initializeDoc();
+
+    let sheet = doc.sheetsByTitle[sheetName];
+
+    if (!sheet) {
+      // Create new worksheet (tab) if it doesn't exist
+      sheet = await doc.addSheet({
+        title: sheetName,
+        headerValues: this.getSheetHeaders(sheetName),
+      });
+    } else {
+      // Always set headers for existing sheets to ensure they're correct
+      const headers = this.getSheetHeaders(sheetName);
+      await sheet.setHeaderRow(headers);
+    }
+
+    return sheet;
+  }
+
+  private getSheetHeaders(sheetName: string): string[] {
+    if (sheetName === "Pitch Submissions") {
+      return [
+        "Submission ID",
+        "Full Name",
+        "Course and Year",
+        "CIT ID",
+        "Phone Number",
+        "Personal Email",
+        "Type of Pitch",
+        "About Pitch",
+        "Pen Name",
+        "Files",
+        "Links",
+        "Submitted At",
+      ];
```

Comment:
Pa butang ani ug constant file please, everything here e usa much better siguro if `sheets-constants.ts` for the constant name ikaw na bahala dapat readable. 
```suggestion
      return [
        "Submission ID",
        "Full Name",
        "Course and Year",
        "CIT ID",
        "Phone Number",
        "Personal Email",
        "Type of Pitch",
        "About Pitch",
        "Pen Name",
        "Files",
        "Links",
        "Submitted At",
      ];
```

Reply:
Never mind the suggested change, misclick rana